File: examples/bert.rs
use embed_anything::config::{BertConfig, EmbedConfig, JinaConfig};
use embed_anything::embed_directory;
use std::{path::PathBuf, time::Instant};

fn main() {
    let now = Instant::now();

    let bert_config = BertConfig {
        model_id: Some("sentence-transformers/all-MiniLM-L12-v2".to_string()),
        revision: None,
        chunk_size: Some(100),
        batch_size: Some(32),
    };

    let jina_config = JinaConfig {
        model_id: Some("jinaai/jina-embeddings-v2-base-en".to_string()),
        revision: None,
        chunk_size: Some(100),
        batch_size: Some(32),
    };

    let config = EmbedConfig {
        // bert: Some(bert_config),
        jina: Some(jina_config),
        ..Default::default()
    };

    let _out = embed_directory(
        PathBuf::from("test_files"),
        "Jina",
        Some(vec!["pdf".to_string()]),
        Some(&config),
        None,
    )
    .unwrap();

    let elapsed_time = now.elapsed();
    println!("Elapsed Time: {}", elapsed_time.as_secs_f32());
}

File: examples/bert.py
import re
from typing import Dict, List
import uuid
import time
import embed_anything
from embed_anything import EmbedData, EmbeddingModel, TextEmbedConfig, WhichModel
from embed_anything.vectordb import Adapter
import os
from time import time


# model = EmbeddingModel.from_pretrained_hf(
#     WhichModel.Bert, model_id="sentence-transformers/all-MiniLM-L12-v2"
# )

model = EmbeddingModel.from_pretrained_cloud(
    WhichModel.OpenAI, model_id="text-embedding-3-small"
)
config = TextEmbedConfig(chunk_size=256, batch_size=32, splitting_strategy = "semantic")

start = time()
data = embed_anything.embed_file("test_files/bank.txt", embeder=model, config=config)

for d in data:
    print(d.text)
    print("---"*20)

# data: list[EmbedData] = embed_anything.embed_directory(
#     "test_files", embeder=model, config=config
# )

end = time()

# for i, d in enumerate(data):
#     print(f"Chunk {i+1}\n")
#     print("-" * 100)

#     print(d.text)
#     print("Metadata\n")
#     print(d.metadata)
#     print("-" * 100)
#     print("\n")

print(f"Time taken: {end - start} seconds")
